dspy/clients/embedding.py

```diff
@@ -41,21 +42,27 @@ class Embedding:
 
         ```python
         import dspy
+        import numpy as np
 
         def my_embedder(texts):
             return np.random.rand(len(texts), 10)
 
-        embedder = dspy.Embedding(my_embedder)
+        embedder = dspy.Embedding(embedding_function=my_embedder)
         embeddings = embedder(["hello", "world"])
 
         assert embeddings.shape == (2, 10)
         ```
     """
 
-    def __init__(self, model):
-        self.model = model
+    def __init__(self, embedding_model: Union[str, Callable[[List[str]], List[List[float]]]] = 'text-embedding-ada-002', embedding_function: Optional[Callable[[List[str]], List[List[float]]]] = None):
```

Comment:
Why do we need this change? It's a common pattern to have model be string for hosted model, and callable for custom model, as most model instances are callable themselves. 

Btw, if we add type annotation, we shouldn't assume the arg type of the callable, which could be easily violated by users' custom function.